promql/ast.go

```diff
@@ -171,15 +171,31 @@ type VectorSelector struct {
 	metrics   map[model.Fingerprint]metric.Metric
 }
 
-func (e *AggregateExpr) Type() model.ValueType  { return model.ValVector }
-func (e *Call) Type() model.ValueType           { return e.Func.ReturnType }
+// Type implements Node.
+func (e *AggregateExpr) Type() model.ValueType { return model.ValVector }
+
+// Type implements Node.
+func (e *Call) Type() model.ValueType { return e.Func.ReturnType }
+
+// Type implements Node.
 func (e *MatrixSelector) Type() model.ValueType { return model.ValMatrix }
-func (e *NumberLiteral) Type() model.ValueType  { return model.ValScalar }
-func (e *ParenExpr) Type() model.ValueType      { return e.Expr.Type() }
-func (e *StringLiteral) Type() model.ValueType  { return model.ValString }
-func (e *UnaryExpr) Type() model.ValueType      { return e.Expr.Type() }
+
+// Type implements Node.
+func (e *NumberLiteral) Type() model.ValueType { return model.ValScalar }
+
+// Type implements Node.
+func (e *ParenExpr) Type() model.ValueType { return e.Expr.Type() }
+
+// Type implements Node.
+func (e *StringLiteral) Type() model.ValueType { return model.ValString }
+
+// Type implements Node.
```

Comment:
Also code obfuscation – those type of "implements" comments might make partially sense. In general they seem off because they turn Go's implicit interface back into Java style where a type has to specify what it implements.

We just went from a compact representation of which node type has which value type to the same thing, just messier and with a useless comment.